### PR TITLE
Support custom VIRTUAL_ENV for layout_python

### LINKED ISSUE #616: Support VIRTUAL_ENV in layout_pyvenv
I was assuming layout_pyvenv (which uses venv) would also respect the `VIRTUAL_ENV` environment variable, like layout_venv does.

To make this happen, I adjusted the `layout_pyvenv` bash code listed on the [Python wiki page](https://github.com/direnv/direnv/wiki/Python).

```bash
layout_pyenv() {
  unset PYENV_VERSION
  # Because each python version is prepended to the PATH, add them in reverse order
  for ((j = $#; j >= 1; j--)); do
    local python_version=${!j}
    local pyenv_python=$(pyenv root)/versions/${python_version}/bin/python
    if [[ ! -x "$pyenv_python" ]]; then
      log_error "Error: $pyenv_python can't be executed."
      return 1
    fi

    unset PYTHONHOME
    local ve=$($pyenv_python -c "import pkgutil; print('venv' if pkgutil.find_loader('venv') else ('virtualenv' if pkgutil.find_loader('virtualenv') else ''))")

    case $ve in
      "venv")
        if [[ -n $VIRTUAL_ENV ]]; then
          VIRTUAL_ENV=$(realpath "${VIRTUAL_ENV}")
        else
          VIRTUAL_ENV=$(direnv_layout_dir)/python-$python_version
        fi
        export VIRTUAL_ENV
        if [[ ! -d $VIRTUAL_ENV ]]; then
          $pyenv_python -m venv "$VIRTUAL_ENV"
        fi
        PATH_add "$VIRTUAL_ENV"/bin
        ;;
      "virtualenv")
        layout_python "$pyenv_python"
        ;;
      *)
        log_error "Error: neither venv nor virtualenv are available to ${pyenv_python}."
        return 1
        ;;
    esac

    # e.g. Given "use pyenv 3.6.9 2.7.16", PYENV_VERSION becomes "3.6.9:2.7.16"
    [[ -z "${PYENV_VERSION-}" ]] && PYENV_VERSION=$python_version || PYENV_VERSION="${python_version}:$PYENV_VERSION"
  done

  export PYENV_VERSION
}
```

I wanted to submit a pull request to make this the default behaviour in direnv, but the `load_pyvenv` in [stdlib.sh](https://github.com/direnv/direnv/blob/a47eb32792d8a231b9df1d1810bff135aa6ee9e4/stdlib.sh#L586) makes use of `layout_python` and thus presumably still uses virtualenv instead of venv?

### PR DESCRIPTION
**Pull request by kinsei0916**
This PR allows users to specify `VIRTUAL_ENV` environment variable for `layout_python` and `layout_pyenv`.
The behavior is along with the [wiki page](https://github.com/direnv/direnv/wiki/Python#for-versions-earlier-than-v2210).
Usage:
```
export VIRTUAL_ENV=.venv
layout pyenv 3.10.1
```
Fix #616

### CODE REVIEW

**Comment by zimbatm** [comment]
Do you mind adding some tests as well? The function is getting pretty large and I am not confident it will work in all scenarios.

**Comment by kinsei0916** [comment]
Thank you for the reply. I will try that.

**Comment by PhilipVinc** [comment]
bump
this would be nice

**Comment by zimbatm** [comment]
the macOS CI is failing. It looks like venv is missing there?

**Comment by kinsei0916** [comment]
Seems to be fixed now. Can you run CI again?

File: stdlib.sh [hunk 1]
```diff
@@ -814,7 +815,13 @@ layout_python() {
       return 1
     fi
 
-    VIRTUAL_ENV=$(direnv_layout_dir)/python-$python_version
+    if [[ -n $VIRTUAL_ENV ]]; then
```
Comment (zimbatm) [inline stdlib.sh:818]:
```suggestion
    if [[ -n "${VIRTUAL_ENV:-}" ]]; then
```
better handle `set -u`

**Review by zimbatm — APPROVED** [review]

**Pull request merged**